File: app/src/androidTest/java/com/mbmc/template/TestApplication.java
package com.mbmc.template;

import com.mbmc.template.di.component.DaggerTestApplicationComponent;
import com.mbmc.template.di.component.TestApplicationComponent;
import com.mbmc.template.di.component.UiComponent;


public class TestApplication extends Application {

    private TestApplicationComponent applicationComponent;


    @Override
    public void onCreate() {
        super.onCreate();

        applicationComponent = DaggerTestApplicationComponent.create();
    }

    @Override
    public UiComponent getUiComponent() {
        return applicationComponent.testUiComponent();
    }

}

File: presentation/src/androidTest/java/com/mbmc/template/TestApplication.kt
package com.mbmc.template

import com.mbmc.template.di.ApplicationComponent
import com.mbmc.template.di.DaggerApplicationComponent
import com.mbmc.template.di.TestApiModule

class TestApplication : Application() {

    override fun onCreate() {
        super.onCreate()
    }

    override fun getComponent(): ApplicationComponent {
        return DaggerApplicationComponent.builder()
            .apiModule(TestApiModule())
            .build()

    }

}